### DISCUSSION TITLE
Matter 1.3 - support for energy monitoring

### DISCUSSION THREAD

**Discussion by ss89** [arendst/Tasmota#21410] [Ideas]
### PROBLEM DESCRIPTION
I want my tasmota devices to report energy consumption measured by an e.g. bl0937 sensor

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Nous A8T
- [x] Tasmota binary firmware version number used: 14.0.0
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: came preflashed with 13.x, updated via tasmota devices web ui
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
20:15:33.342 CMD: Backlog Template; Module; GPIO 255
20:15:33.394 RSL: RESULT = {"NAME":"NOUS A8T","GPIO":[1,1,320,1,32,1,1,1,1,224,2624,1,1,1,1,1,0,1,1,1,0,1,2656,2720,0,0,0,0,1,1,1,1,1,0,0,1],"FLAG":0,"BASE":1}
20:15:33.623 RSL: RESULT = {"Module":{"0":"NOUS A8T"}}
20:15:33.835 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"320":"Led_i1"},"GPIO3":{"0":"None"},"GPIO4":{"32":"Button1"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"224":"Relay1"},"GPIO14":{"2624":"HLWBL SEL_i"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO24":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"2656":"HLWBL CF1"},"GPIO27":{"2720":"BL0937 CF"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
commission tasmota device with home assistant, see there is a relay and cpu temperature - that's it. No energy monitoring.

### EXPECTED BEHAVIOUR
Allow to add energy monitoring in tasmotas matter menu (maybe include in the docs how to refer the sensor properly too?)

### SCREENSHOTS
not needed

### ADDITIONAL CONTEXT
-

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by s-hadinger** [comment]
This is on the roadmap

**Reply by MrRomo** [reply]
Hi, how it's goin the matter monitor ?

**Comment by nikito7** [comment]
Home Assistant:

Add support for Matter 1.3 power/energy sensors
Especially the energy sensors are great to use with the energy dashboard.

Waiting for tasmota

**Reply by simonostendorf** [reply]
I would also like to see this

**Comment by lukejagger** [comment]
Would also be good to have energy monitoring enabled for 'Remote Tasmotas' (ESP8266's accessed through an ESP32 Matter enabled Tasmota device) and ESP32 smartplugs with Matter enabled Tasmota too.

**Reply by s-hadinger** [reply]
It's still of limited usefulness until Apple/Google/Alexa support energy montiroring. Until then it will probably stay on hold

**Reply by ss89** [reply]
Maybe i'm not getting it, but wouldn't that still be useful for Home Assistant and other Home Automation platforms?

**Comment by stefxx** [comment]
I would like that as well!

**Comment by Hl2run** [comment]
Any idea please about the Matter 1.3 implementation? Thanks 🙏

**Reply by s-hadinger** [reply]
Yes I will soon expand capabilities, but I'm focusing now on enhancing the configuration interface so we can onboard richer configuration.

**Comment by pilnikov** [comment]
<img width="484" height="555" alt="image" src="https://github.com/user-attachments/assets/8a64f46f-de20-4b85-a208-198f66b32743" />

++ in HA - all ok!!! But in matter config :( .
